crates/sui-core/src/execution_engine.rs

```diff
@@ -574,7 +574,7 @@ fn pay_sui<S>(
 
     #[cfg(debug_assertions)]
     {
-        assert_eq!(total_coins - merged_coin.value(), total_amount)
+        assert_eq!(total_coins - merged_coin.value(), _total_amount)
```

Comment:
I think it would be easier to just use `debug_assert_eq` macro here. No need to rename variable in that case

Reply:
oh great to know, this is better indeed, thanks!